messaging-activemq/src/main/java/org/wildfly/extension/messaging/activemq/ActiveMQServerService.java

```diff
@@ -304,16 +302,11 @@ public synchronized void start(final StartContext context) throws StartException
                 for(final Map.Entry<String, DiscoveryGroupConfiguration> entry : discoveryGroups.entrySet()) {
                     final String name = entry.getKey();
                     final String key = "discovery" + name;
-                    DiscoveryGroupConfiguration config = null;
+                    final DiscoveryGroupConfiguration config;
                     if (jgroupFactories.containsKey(key)) {
                         ChannelFactory channelFactory = jgroupFactories.get(key);
                         String channelName = jgroupsChannels.get(key);
-                        JChannel channel = channels.get(channelName);
-                        if (channel == null) {
-                            channel = (JChannel) channelFactory.createChannel(channelName);
-                            channels.put(channelName, channel);
-                        }
-                        config = DiscoveryGroupAdd.createDiscoveryGroupConfiguration(name, entry.getValue(), channel, channelName);
+                        config = DiscoveryGroupAdd.createDiscoveryGroupConfiguration(name, entry.getValue(), channelFactory, channelName, channels);
```

Comment:
@jmesnil Question - the intention of the previous logic seems to be that only 1 JGroups channel would be created for every unique channel name between the broadcast group and discovery groups.
This change moves the creation of the channel to the endpoint and defers the management of channel sharing between multiple endpoints to the JChannelManager.  I think that makes sense.  Ideally, a JChannel should be closed by the same object that created it.
However, this seems to render the Map<String, JChannel> in ActiveMQServerService obsolete.  Can we remove this?  The channels created by discovery group endpoints are added to this map and removed on close, but I don't see anything that ever reads from this map.